regression/check_style.sh

```diff
@@ -69,91 +86,194 @@ if [[ ! ${cf} ]]; then
-  cmd='${gcf} --binary ${cf} -f --diff --extensions hh,cc develop'
-  echo "Running..."
-  echo "   ${gcf} --binary ${cf} -f --diff --extensions hh,cc develop"
-  echo " "
-  result=`eval $cmd`
-  allok=`echo $result | grep -c "did not modify"`
-  # 2nd chance (maybe there are no files to check)
-  if test $allok = 0; then
-    allok=`echo $result | grep -c "no modified files"`
-  fi
+# don't actually modify the files (compare to branch 'develop')
+cmd="${gcf} --binary ${cf} -f --diff --extensions hh,cc develop"
+run "${cmd}" &> $patchfile_c
 
-  if test $allok = 1; then
-    echo "PASS: Changes conform to this project's style requirements."
+# if the patch file has the string "no modified files to format", the check passes.
+if [[ `grep -c "no modified files" ${patchfile_c}` != 0 ]]; then
+  echo "PASS: Changes to C++ sources conform to this project's style requirements."
+else
+  foundissues=1
+  echo    "FAIL: some C++ files do not conform to this project's style requirements:"
+  # Modify files, if requested
+  if [[ ${fix_mode} == 1 ]]; then
+    echo -e "      The following patch has been applied to your file.\n"
+    run "git apply $patchfile_c"
+    cat $patchfile_c
   else
-    echo "FAIL: some files do not conform to this project's style requirements:"
-    echo " "
-    # rerun the command to capture color output.
-    eval $cmd
-    exit 1
+    echo -e "      run ${0##*/} with option -f to automatically apply this patch.\n"
+    cat $patchfile_c
   fi
+fi
+rm -f $patchfile_c
 
 ##---------------------------------------------------------------------------##
-## Fix mode
-##   no options --> fix the files by running clang-format
-##   -d | -n    --> print diff of required changes.
+## Check mode (Test F90 code indentation with emacs and bash)
 ##---------------------------------------------------------------------------##
 
-else
+# Defaults ----------------------------------------
+EMACS=`which emacs`
+if [[ $EMACS ]]; then
+  EMACSVER=`$EMACS --version | head -n 1 | sed -e 's/.*Emacs //'`
+  if [[ `version_gt "24.0.0" $EMACSVER`  ]]; then
+    echo "WARNING: Your version of emacs is too old. Expecting v 24.0+."
+    echo "         pre-commit-hook partially disabled (f90 indentation)"
+    unset EMACS
+  fi
+fi
 
-  if test ${diff_mode} = 1; then
-    cmd='${gcf} --binary ${cf} -f --diff --extensions hh,cc develop'
-    result=`eval $cmd`
-    echo "The following non-conformances were discovered. Rerun without -d|-n to"
-    echo "automatically apply these changes:"
-    echo " "
-    # rerun command to capture color output.
-    eval $cmd
+# staged files
+modifiedfiles=`git diff --name-only --cached`
+# unstaged files
+modifiedfiles="${modifiedfiles} `git diff --name-only`"
+# all files
+modifiedfiles=`echo ${modifiedfiles} | sort -u`
+
+# file types to parse. Only effective when PARSE_EXTS is true.
+FILE_EXTS=".f90 .F90"
+
+# file endings for files to exclude from parsing when PARSE_EXTS is true.
+FILE_ENDINGS="_f.h _f77.h _f90.h"
```

Comment:
They are not strictly required, but I wanted to provide an example of this feature since the `matches_extension` bash function looks for this variable.